### PR TITLE
Replace tribute with text-expander-element for textarea

### PR DESCRIPTION
**Pull request by silverwind**
The completion popup now behaves now much more as expected than before for the raw textarea:
- You can press <kbd>Tab</kbd> or <kbd>Enter</kbd> once the completion popup is open to accept the selected item
- The menu does not close automatically when moving the cursor
- When you delete text, previously correct suggestions are shown again
- If you delete all text until the opening char (`@` or `:`) after applying a suggestion, the popup reappears again
- Menu UI has been improved

Screenshot 2023-04-07 at 19 43 42

### CODE REVIEW

File: web_src/js/features/comp/ComboMarkdownEditor.js [hunk 1]
```diff
@@ -83,6 +82,40 @@ class ComboMarkdownEditor {
...
+    expander?.addEventListener('text-expander-change', ({detail: {key, provide, text}}) => {
+      if (key === ':') {
+        const ul = document.createElement('ul');
+        ul.classList.add('suggestions');
+
+        const matches = [];
+        for (const name of emojiKeys) {
+          if (name.includes(text)) {
+            matches.push(name);
+            if (matches.length > 5) break;
```
Comment (silverwind) [inline web_src/js/features/comp/ComboMarkdownEditor.js]:
```suggestion
            if (matches.length > 4) break;
```

**Comment by silverwind** [comment]
Tangentially related change: Remove fomantic's `transition: 1s color` on `input` and `textarea`.

**Comment by silverwind** [comment]
Mentions implemented, so this is ready for review.

Screenshot 2023-04-07 at 23 11 38

Changes to previous tribute include:

- Full name is now hidden when it matches the name case-sensitively (Above screenshot has it only for showcase purpose).
- Tab key now triggers competion as well

**Review by delvh — APPROVED** [review]
Have tested it.
It even fixes the most annoying bugs I've noticed previously, i.e. that the popup suddenly disappears because you _dared_ to move your cursor :relieved:

**Comment by delvh** [comment]
> Tab key now triggers completion as well

For some reason, I haven't been able to see that.
When I press <kbd>Tab</kbd> while inside the textbox, the textbox loses focus.
Or how is that meant?

**Comment by silverwind** [comment]
> > Tab key now triggers completion as well
> 
> For some reason, I haven't been able to see that. When I press Tab while inside the textbox, the textbox loses focus. Or how is that meant?

The key sequence `:`, `<Tab>` inserts 👍 for me. Not for you?

**Comment by delvh** [comment]
Ah, that's what you mean.
Yep, then I've noticed this as well already.

**Comment by silverwind** [comment]
Yeah, it's something I wanted for a long time, but it was not possible with tribute without forking it.

**Review by lunny — APPROVED** [review]

**Comment by lunny** [comment]
I can use `Ente` or `Tab` to select the item now. It's great.

**Comment by silverwind** [comment]
.

**Comment by silverwind** [comment]
.

**Comment by silverwind** [comment]
.

**Pull request merged**